tests/Integration/Queue/ProcessorTest.php

```diff
@@ -204,6 +205,33 @@ public function test_processRequestSets_ShouldReturnOnlyValidRequestSetsInCaseTh
         $this->assertCount(2, $requestSet5->getRequests());
     }
 
+    public function test_processRequestSets_ShouldCatchTypeError()
+    {
+        $tracker = $this->createTracker();
+        $queuedRequestSets = [
+            $requestSet1 = $this->buildRequestSet(5),
+            $requestSet2 = $this->buildRequestSet(2),
+        ];
+
+        // Make one of the requests have an unexpected value type
+        $requestParams = $requestSet2->getRequests()[1]->getRawParams();
+        // The uadata field is expected to be a JSON string and not an array. It will throw a TypeError during decoding
+        $requestParams['uadata'] = [];
+        $requestSet2->setRequests([$requestSet2->getRequests()[0], new Tracker\Request($requestParams)]);
+
+        $this->acquireAllQueueLocks();
+        $requestSetsToRetry = $this->processor->processRequestSets($tracker, $queuedRequestSets);
+
+        // I couldn't find a param other than uadata that could throw a TypeError and it's not used in older versions
+        $isOlderMatomo = version_compare(Version::VERSION, '4.12.0', '>');
+        $expectedSets = $isOlderMatomo ? [$requestSet1, $requestSet2] : [];
```

Comment:
@snake14 if you extend this test to re run, $expectedSets should be [$requestSet1] for newer version right ? If yes can you include that here too ?

Reply:
Thank you @AltamashShaikh . Making that change made me realise that I misnamed the variable.